File: PuntoC/Punto2D.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PuntoC
{
    public class Punto2D:Punto
    {
        protected int _y;

       
        public int Y
        {
            get { return _y; }
            set { _y = value; }
        }

        public Punto2D()
        {
            _x = 0;
            _y = 0;
        }

        public Punto2D(int x, int y)
        {
            _x = x;
            _y = y;
        }
        public override string ToString()
        {
            return "(" + _x + "," + _y + ")";
        }
        public double Distancia(Punto2D p)
        {
            double resultado = 0;
            double xp = (_x - p.X) * (_x - p.X);
            double yp = (_y - p.Y)* (_y - p.Y);
            resultado = Math.Sqrt(xp+yp);
            return resultado;
        }
    }
}

File: PuntoVB/Punto2D.vb
Public Class Punto2D
    Inherits Punto


    Protected y1 As Integer

    Public Sub New()
        x1 = 0
        y1 = 0
    End Sub
    Public Sub New(x As Integer, y As Integer)
        x1 = x
        y1 = y
    End Sub



    Public Property Y() As Integer
        Get
            Return y1
        End Get

        Set(ByVal value As Integer)
            y1 = value
        End Set
    End Property

    Public Overloads Function Distancia(p As Punto2D) As Double
        Dim resultado As Double = 0
        Dim xp = (x1 - p.X) * (x1 - p.X)
        Dim yp = (y1 - p.Y) * (y1 - p.Y)
        resultado = Math.Sqrt(xp + yp)
        Return resultado
    End Function
    Public Overrides Function ToString() As String
        Return "(" + x1 + "," + y1 + ")"
    End Function
End Class
